output/create_kubevirt_yaml.ml

```diff
@@ -209,6 +210,15 @@ let create_kubevirt_yaml source inspect
         Assoc [ "networkName", String vnet; "name", String ("net_" ^ vnet) ]
     ) target_nics in
 
+  (* The spec -> preference section is where we can pass the guest type.
+   * However not, unfortunately, using libosinfo, but using a made
+   * up set of strings which we have to map:
+   * https://github.com/kubevirt/common-instancetypes
+   *)
+  Option.iter (
+    fun name -> List.push_back spec_preference ("name", String name)
+  ) (kubevirt_instancetype guestcaps inspect);
+
```

Comment:
I think there's a slight misunderstanding here, we use preferences to provide various defaults in the eventual VMI. The example output in the PR description is from the VMI, can v2v also provide VM output?

Reply:
I'm not sure of the terminology here, but virt-v2v converted VMs are always pets (since they come from VMware) and each has its own special memory size, number of CPUs, devices, etc, and we are attempting to express that in the yaml file we generate.

Reply:
If they are pets with things such as disk buses, interface models, rng etc already provided then you don't need to use preferences at all. There's zero benefit if the `VirtualMachine` already has all of these choices at the moment.

FWIW looking more at the code it's currently generating a `VirtualMachineInstance` (running VM)  that don't even support the use of preferences anyway. `VirtualMachine` (persistent VM) should likely be the output of this tool.